### PR TITLE
[RHPAM-3681] - [1.5.x] - Introduce BuildContext

### PR DESCRIPTION
**Pull request by vaibhavjainwiz**
Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [ ] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [ ] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [ ] Pull Request contains a link to the JIRA issue
- [ ] Pull Request contains a description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/master/RELEASE_NOTES.md) entry regarding this change

### CODE REVIEW

**Comment by radtriste** [comment]
cherry-pick: https://github.com/kiegroup/kogito-operator/pull/917

**Review by radtriste — APPROVED** [review]

**Review by spolti — APPROVED** [review]

File: core/kogitobuild/buid_config_test.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 Red Hat, Inc. and/or its affiliates
```
Comment (Kevin-Mok) [inline core/kogitobuild/buid_config_test.go:1]: This file name has a minor typo. :sweat_smile:

Comment (spolti) [reply core/kogitobuild/buid_config_test.go:1]: Nice catch Kevin, do you mind fixing it?

Comment (Kevin-Mok) [reply core/kogitobuild/buid_config_test.go:1]: Sent PR's for [`master`](https://github.com/kiegroup/kogito-operator/pull/923) and [`1.5.x`](https://github.com/kiegroup/kogito-operator/pull/922) branch.

**Pull request merged**